posixlib/src/main/scala/scala/scalanative/posix/poll.scala

```diff
@@ -0,0 +1,82 @@
+package scala.scalanative
+package posix
+
+import scalanative.native._
+
+object pollNfds {
+
+  type nfds_t = CUnsignedLongInt
+
+  // Keep the breakage in the nfds_t abstraction confined to here.
+  // The following code will work on both 64 and 32 bit machines.
+  //
+  // posix FOPEN_MAX is currently defined in scalanative wrap.c
+  // as an unsigned int. So any scala Long with bits greater than #31
+  // set are going to fail anyway. Some/most systems will probably
+  // exceed FOPEN_MAX earlier than that.
+  //
+  // Revisit & simplify when better 32/64 bit support arrives.
+
+  @inline
+  def toNfsd_t(in: Int): nfds_t = in.toUInt
+
+  def toNfsd_t(in: ULong): nfds_t = {
+    val uIntMax = UInt.MaxValue
+    if (in > uIntMax) {
+      throw new Exception(s"pollNfds: too many nfds: ${in}  > ${uIntMax}")
+    } else {
+      // may still fail later if greater than FOPEN_MAX. Let code
+      // more closely concerned test for that.
+      // On 64 bit machines UInt will get silently promoted match the
+      // ULong nfsd_t on that machine, so types work out.
+      in.toUInt // truncation is now harmless.
+    }
+  }
```

Comment:
This looks like a hack to be honest. We should assume we only support 64-bit until 32-bit is officially merged.

Reply:
Sir, you do  that section great honor by calling it a hack ;-)

Put into motion by your spot on comment, I removed that section entirely
and focused on the issue I was trying to express.  

Upon further study, I figured out that  this is/was not a 32/64 bit issue, but a signed-Int to unsigned-Int conversion issue.  Especially how to make it easier for the most common
user to get it right:Least Astonishment.  

I put a TL;DR comment just after the declaration of the poll() method, showing the
intended usage.

Then I put a longer comment describing the rationale for the suggested usage.
The Haskell school would say that comment is excessive and should be deleted.
I put it there because I could barely figure out the reasoning this time and am
pretty sure I would spend days figuring it out in a year.  Others may be sharper
but I hope to be kind to all.